### PR TITLE
Alexcmonnet/validating yaml with 386

### LINKED ISSUE #386: Validating yaml with malformed ext does not work properly
I've been beating my head against the validator for a while now.  All it was providing was a general error and a line reference nowhere close to where the issue was.  It appears the language context is trying to apply the `ext` definition in the model file being validated rather than just validate the contents.

To repeat the issue define a enum incorrectly (in this case using the old `dataExt` rather than `schemaExt`):
```
ext:
  name: addReportToRoot
  type: root
  dataExt:
    add:
      - name: report
        type: Report
```
When you run `aac validate file-with-invalid-aac.yaml` you get the following resopnse:
```
validate: general_failure

An unrecognized error occurred:/workspace/AaC/python/src/aac/cli/builtin_commands/validate/validate_impl.py line 20 message: 'NoneType' object has no attribute 'get'
```
The expected behavior would be a message stating that a defined field in the provided yaml file is incorrect.

### PR DESCRIPTION
**Pull request by Coffee2Bits**
Closes #386

* Added error handling for malformed extensions

### CODE REVIEW

File: python/src/aac/validate/_validate.py [hunk 1]
```diff
@@ -43,10 +44,13 @@ def validated_source(source: str) -> Generator[ValidationResult, None, None]:
 
 def _with_validation(user_definitions: list[Definition]) -> ValidationResult:
     result = _validate_definitions(user_definitions)
-    if result.is_valid:
-        return result
-    else:
-        raise ValidationError("Failed to validate content with errors:", *result.messages)
+    try:
```
Comment (cvchaparro) [inline python/src/aac/validate/_validate.py:47]: I couldn't leave it as a suggestion but I think this `try` should go around the `result = ...` part, too. I don't see any way in which this `if` statement would result in a `LanguageError`.

**Review by cvchaparro — CHANGES_REQUESTED** [review]
Looks good, there was just one thing I noticed that I think needs to be changed.

File: python/src/aac/validate/_validate.py (hunk 1 above)
Comment (Coffee2Bits) [reply python/src/aac/validate/_validate.py:47]: Good catch, thanks!

**Review by cvchaparro — APPROVED** [review]
Nice work!

**Pull request merged**